### PR TITLE
New Motoman BMDA3 URDF

### PR DESCRIPTION
**Pull request by phvass**
Files for URDF are included. xacro still needed.

### CODE REVIEW

**Comment by shaun-edwards** [comment]
+1

File: motoman_bmda3_support/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+catkin_package()
+
+if (CATKIN_ENABLE_TESTING)
+  find_package(roslaunch REQUIRED)
+  roslaunch_add_file_check(test/launch_test.xml)
+endif()
+
+foreach(dir config launch meshes urdf)
+   install(DIRECTORY ${dir}/
```
Comment (gavanderhoorn) [inline motoman_bmda3_support/CMakeLists.txt:15]: style: indentation: 2 spaces per level.

File: motoman_bmda3_support/config/joint_names_bmda3.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+controller_joint_names: ['joint_s', 'joint_l', 'joint_u', 'joint_r', 'joint_b', 'joint_t']
```
Comment (gavanderhoorn) [inline motoman_bmda3_support/config/joint_names_bmda3.yaml:1]: These names do not match those in the URDF.

File: motoman_bmda3_support/launch/robot_state_visualize_bmda3.launch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!--
```
Comment (gavanderhoorn) [inline motoman_bmda3_support/launch/robot_state_visualize_bmda3.launch:1]: style: indentation.

File: motoman_bmda3_support/package.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<package>
```
Comment (gavanderhoorn) [inline motoman_bmda3_support/package.xml:1]: @shaun-edwards: we should really fix the templates in `industrial_robot_pkg_gen` :) (indentation, maintainer without name, pdf does not exist, `bmda3` should be properly capitalised).

File: motoman_bmda3_support/urdf/bmda3.urdf [hunk 5]
```diff
@@ -0,0 +1,689 @@
...
+  name="motoman_bmda3_support">
+  <link
+    name="base_link">
+    <visual>
+      <origin
+        xyz="0 0 0"
+        rpy="0 0 0" />
+      <geometry>
+        <mesh
+          filename="package://motoman_bmda3_support/meshes/bmda3/visual/base_link.STL" />
```
Comment (gavanderhoorn) [inline motoman_bmda3_support/urdf/bmda3.urdf:11]: uppercase extensions

File: motoman_bmda3_support/urdf/bmda3.urdf [hunk 6]
```diff
@@ -0,0 +1,689 @@
+<robot
+  name="motoman_bmda3_support">
```
Comment (gavanderhoorn) [inline motoman_bmda3_support/urdf/bmda3.urdf:2]: `name` should be `motoman_bdma3`.

**Comment by gavanderhoorn** [comment]
The urdf needs to be fixed (joint names, exploit `left`, `right` symmetry). Address in #2?

File: motoman_bmda3_support/config/joint_names_bmda3.yaml (hunk 2 above)
Comment (shaun-edwards) [inline motoman_bmda3_support/config/joint_names_bmda3.yaml:1]: @phvass, please let IPA know they will have to fix these (we need to find a good pattern so we can add this to the auto-generator tool)

File: motoman_bmda3_support/urdf/bmda3.urdf (hunk 6 above)
Comment (shaun-edwards) [inline motoman_bmda3_support/urdf/bmda3.urdf:2]: @phvass, please fix in latest PR #3

File: motoman_bmda3_support/config/joint_names_bmda3.yaml (hunk 2 above)
Comment (gavanderhoorn) [inline motoman_bmda3_support/config/joint_names_bmda3.yaml:1]: @shaun-edwards: not sure that is really needed: the utility of this file is becoming questionable. Especially in the case of multi group cells and namespacing users will have to update / edit these anyway, and the file isn't used by anything but the launch files we supply in the support packages (the MoveIt configurations by default use their own version of the flle).

**Pull request merged**